packages/fxa-auth-server/fxa-oauth-server/test/api.js

```diff
@@ -4657,4 +4684,126 @@ describe('/v1', function() {
       assert.isFalse(result.active);
     });
   });
+
+  describe('JWT access token', () => {
+    it('succeeds', async () => {
+      const jwtClient = clientByName('JWT Client');
+      assert(jwtClient.canGrant); //sanity check
+      const clientId = jwtClient.id;
+      mockAssertion().reply(200, VERIFY_GOOD);
+
+      const authorizationResult = await Server.api.post({
+        url: '/authorization',
+        payload: authParams({
+          client_id: clientId,
+        }),
+      });
+
+      assert.equal(authorizationResult.statusCode, 200);
+
+      const tokenResult = await Server.api.post({
+        url: '/token',
+        payload: {
+          client_id: clientId,
+          client_secret: secret,
+          code: authorizationResult.result.code,
+        },
+      });
+
+      assert.equal(tokenResult.statusCode, 200);
+      assertSecurityHeaders(tokenResult);
+      assert.ok(tokenResult.result.access_token);
+      assert.include(tokenResult.result.access_token, '.');
+      assert.strictEqual(tokenResult.result.token_type, 'bearer');
+      assert.ok(tokenResult.result.auth_at);
+      assert.ok(tokenResult.result.expires_in);
+      assert.equal(tokenResult.result.scope, 'a');
+      assert.isUndefined(tokenResult.result.keys_jwe);
+
+      const verifyResult = await Server.api.post({
+        url: '/verify',
+        payload: {
+          token: tokenResult.result.access_token,
+        },
+      });
+
+      assert.equal(verifyResult.statusCode, 200);
+
+      const introspectResult = await Server.api.post({
+        url: '/introspect',
+        payload: {
+          token: tokenResult.result.access_token,
+          token_type_hint: 'access_token',
+        },
+      });
+
+      assert.equal(introspectResult.statusCode, 200);
+      assert.isTrue(introspectResult.result.active);
+
+      const destroyResult = await Server.api.post({
+        url: '/destroy',
+        payload: {
+          token: tokenResult.result.access_token,
+        },
+      });
+
+      assert.equal(destroyResult.statusCode, 200);
+
+      const verifyInvalidTokenResult = await Server.api.post({
+        url: '/verify',
+        payload: {
+          token: tokenResult.result.access_token,
+        },
+      });
+
+      assert.equal(verifyInvalidTokenResult.statusCode, 400);
+      assert.equal(verifyInvalidTokenResult.result.errno, 108);
+
+      const introspectInvalidTokenResult = await Server.api.post({
+        url: '/introspect',
+        payload: {
+          token: tokenResult.result.access_token,
+          token_type_hint: 'access_token',
+        },
+      });
+
+      assert.equal(introspectInvalidTokenResult.statusCode, 200);
+      assert.isFalse(introspectInvalidTokenResult.result.active);
+    });
+  });
+
+  describe('Access w/ BrowserID assertion', () => {
+    it('succeeds', async () => {
```

Comment:
I'm not sure "succeeds" gives the right impression here :-P

Maybe "is cleanly rejected" or similar?